### PR TITLE
Add docs for new type check template functions

### PR DESCRIPTION
**Pull request by raman325**
## Proposed change
Documents the following additional functions/filters/tests for templates: https://github.com/home-assistant/core/pull/100807

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/100807
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes 

## Checklist

- [ ] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_docs/configuration/templating.markdown [hunk 1]
```diff
@@ -916,6 +916,23 @@ The numeric functions and filters raise an error if the input is not a valid num
 - Filter `value_one|bitwise_or(value_two)` perform a bitwise or(\|) operation with two values.
 - Filter `ord` will return for a string of length one an integer representing the Unicode code point of the character when the argument is a Unicode object, or the value of the byte when the argument is an 8-bit string.
 
+### Complex type checking
+
+In addition to strings and numbers, Python (and Jinja) support lists, sets, and dictionaries. To help you with testing these types, the following are available as functions, filters, and tests:
```
Comment (emontnemery) [inline source/_docs/configuration/templating.markdown]: support -> supports
```suggestion
In addition to strings and numbers, Python (and Jinja) supports lists, sets, and dictionaries. To help you with testing these types, the following are available as functions, filters, and tests:
```

File: source/_docs/configuration/templating.markdown [hunk 2]
```diff
@@ -916,6 +916,23 @@ The numeric functions and filters raise an error if the input is not a valid num
...
+
+- `is_list(x)` will return whether `x` is a `list` or not (e.g. [1, 2]).
```
Comment (emontnemery) [inline source/_docs/configuration/templating.markdown]: Please rewrite the example ("e.g. [1, 2]") is an example of a list, and does not explain how `is_list` behaves.
Also, please explain how it works when used as a filter or test too.
Same comment for the other `is_xxx`` .

It's mentioned in the core PR, Jinja has a built in function called "mapping", can we mention it here too? Does that also allow use as a function, filter or test?

**Review by emontnemery — CHANGES_REQUESTED** [review]

File: source/_docs/configuration/templating.markdown [hunk 3]
```diff
@@ -916,6 +916,27 @@ The numeric functions and filters raise an error if the input is not a valid num
...
+
+- `x is list` will return whether `x` is a `list` or not (e.g. `[1, 2] is list` will return `True`).
+- `x is set` will return whether `x` is a `set` or not (e.g. `{1, 2} is set` will return `True`).
+- `x is tuple` will return whether `x` is a `tuple` or not (e.g. `(1, 2) is tuple` will return `True`).
+- `x is datetime` will return whether `x` is a `datetime` or not (e.g. `datetime(2020, 1, 1, 0, 0, 0) is datetime` will return `True`).
+- `x is string_like` will return whether `x` is a string, bytes, or bytearray object.
+
+Note that, as of Home Assistant 2023.10.0, Jinja has built in tests for `boolean` (`True`/`False`), `callable` (any function), `float` (a number with a decimal), `integer` (a number without a decimal), `iterable` (a value that can be iterated over such as a `list`, `set`, `string`, or generator), `mapping` (mainly `dict` but also supports other dictionary like types), `number` (`float` or `int`), `sequence` (a value that can be iterated over and indexed such as `list` and `string`), and `string`.
```
Comment (frenck) [inline source/_docs/configuration/templating.markdown]:
```suggestion
Note that, in Home Assistant, Jinja has built-in tests for `boolean` (`True`/`False`), `callable` (any function), `float` (a number with a decimal), `integer` (a number without a decimal), `iterable` (a value that can be iterated over such as a `list`, `set`, `string`, or generator), `mapping` (mainly `dict` but also supports other dictionary like types), `number` (`float` or `int`), `sequence` (a value that can be iterated over and indexed such as `list` and `string`), and `string`.
```

File: source/_docs/configuration/templating.markdown [hunk 4]
```diff
@@ -916,6 +916,27 @@ The numeric functions and filters raise an error if the input is not a valid num
...
+
+### Type conversions
+
+While Jinja natively supports the conversion of an iterable to a `list`, it does not support conversion to `tuple` or `set`. To help you with using these types, you can use the following functions:
```
Comment (frenck) [inline source/_docs/configuration/templating.markdown]:
```suggestion
While Jinja natively supports the conversion of an iterable to a `list`, it does not support conversion to a `tuple` or `set`. To help you with using these types, you can use the following functions:
```

File: source/_docs/configuration/templating.markdown [hunk 5]
```diff
@@ -916,6 +916,27 @@ The numeric functions and filters raise an error if the input is not a valid num
...
+
+- `set(x)` will convert any iterable `x` to a `set` (e.g. `set([1, 2]) == {1, 2}`)
+- `tuple(x)` will convert any iterable `x` to a `tuple` (e.g. `tuple("abc") == ("a", "b", "c")`)
+
+Note that, as of Home Assistant 2023.10.0, to convert a value to a `list`, a `string`, an `int`, or a `float`, Jinja has built in functions with names that correspond to each type.
```
Comment (frenck) [inline source/_docs/configuration/templating.markdown]:
```suggestion
Note that, in Home Assistant, to convert a value to a `list`, a `string`, an `int`, or a `float`, Jinja has built-in functions with names that correspond to each type.
```

**Review by frenck — APPROVED** [review]
Thanks, @raman325 👍 

../Frenck

**Pull request merged**